### PR TITLE
[TW-1402] Doc data file support for performance testing

### PR DESCRIPTION
**Pull request by mark-anthony-postman**
- Break apart existing "Test API performance" page into 3 pages and update links
- Add new "Import data for virtual users" page
- Clarify information about string handling for imported data files (fixes [TW-1428](https://postmanlabs.atlassian.net/browse/TW-1428))

View the beta docs here:
- NOTE: The first 3 pages aren't new content--they are just the existing content broken up into 3 pages; the 4th page about importing data for virtual users is all new content and needs to be reviewed
- [Run a performance test](https://tw-1402-performance-test-data-files.learning.postman-beta.com/docs/collections/performance-testing/testing-api-performance/)
- [View performance test metrics](https://tw-1402-performance-test-data-files.learning.postman-beta.com/docs/collections/performance-testing/performance-test-metrics/)
- [Debug performance test errors](https://tw-1402-performance-test-data-files.learning.postman-beta.com/docs/collections/performance-testing/performance-test-errors/)
- [Inject data into virtual users](https://tw-1402-performance-test-data-files.learning.postman-beta.com/docs/collections/performance-testing/performance-test-data-files/)

[TW-1428]: https://postmanlabs.atlassian.net/browse/TW-1428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/docs/collections/performance-testing/performance-test-errors.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+From the **Errors** tab, you can take the following actions:
+
+* Hover over a point on the graph to view more details.
+* Select a metric in the legend below the graph to show or hide that metric in the graph.
+* To view the error trends for a specific request, select the request in the dropdown list. This information can help you understand which requests are associated with certain types of errors. To change the graph back to the default view, select **Reset**.
+
+<img alt="Viewing top errors" src="https://assets.postman.com/postman-docs/v10/performance-test-filter-errors-v10-15.jpg" width="600px"/>
+
+## Debugging errors
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/performance-test-errors.md:62]:
```suggestion
## Debug errors
```

File: src/pages/docs/collections/performance-testing/performance-test-errors.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+You can view details about errors that occurred during the performance test to help in the debugging process. The **Errors** tab shows the error classes that were encountered during the test and the number of times each error occurred.
+
+Expand an error class below the graph to view all requests that encountered the error. For each request, you can view a list of unique responses (as determined by headers and body). Postman displays the 10 most common responses and shows a count of how often that response occurred.
+
+Select a request to view details about it, including the response, headers, and request body. Use the controls on the **Response** tab to format, copy, or search within the response. Select the name of a request to open it.
+
+<img alt="Viewing error details" src="https://assets.postman.com/postman-docs/v10/performance-test-errors-detail-v10-17.jpg"/>
+
+> Request and responses details for errors are persisted locally during your current Postman session and aren't saved permanently. If you close the current tab or quit Postman, request and response details will no longer be available. You still be able to view error classes and error counts for requests.
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/performance-test-errors.md:72]:
```suggestion
> Request and responses details for errors are persisted locally during your current Postman session and aren't saved permanently. If you close the current tab or quit Postman, request and response details will no longer be available. You will still be able to view error classes and error counts for requests.
```

File: src/pages/docs/collections/performance-testing/performance-test-metrics.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  - type: link
+    name: "Test your API’s performance by simulating real-world traffic with Postman"
+    url: "https://blog.postman.com/postman-api-performance-testing/"
+---
+
+During a [performance test](/docs/collections/performance-testing/testing-api-performance/), all requests are sent from your computer where you are running the Postman desktop app. You can view real-time metrics such as the average response time, error rate, and throughput. You can also customize the performance test graph to focus on the requests or test metrics you're most interested in.
+
+## Contents
+
+* [Viewing performance metrics](#viewing-performance-metrics)
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/performance-test-metrics.md:23]:
```suggestion
* [View performance metrics](#view-performance-metrics)
```

File: src/pages/docs/collections/performance-testing/performance-test-metrics.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+* [Customizing the performance view](#customizing-the-performance-view)
+
+## Viewing performance metrics
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/performance-test-metrics.md:26]:
```suggestion
## View performance metrics
```

File: src/pages/docs/collections/performance-testing/performance-test-metrics.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    name: "Test your API’s performance by simulating real-world traffic with Postman"
+    url: "https://blog.postman.com/postman-api-performance-testing/"
+---
+
+During a [performance test](/docs/collections/performance-testing/testing-api-performance/), all requests are sent from your computer where you are running the Postman desktop app. You can view real-time metrics such as the average response time, error rate, and throughput. You can also customize the performance test graph to focus on the requests or test metrics you're most interested in.
+
+## Contents
+
+* [Viewing performance metrics](#viewing-performance-metrics)
+* [Customizing the performance view](#customizing-the-performance-view)
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/performance-test-metrics.md:24]:
```suggestion
* [Customize the performance view](#customize-the-performance-view)
```

File: src/pages/docs/collections/performance-testing/performance-test-metrics.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+* Select **Run Again** to configure and start the performance test again.
+* To get a link to the performance report, select the share icon <img alt="Share icon" src="https://assets.postman.com/postman-docs/icon-share.jpg#icon" width="16px">. Select **Copy Link** to copy the link to your clipboard. You can share the link with your teammates so they can view the results of the performance test.
+* To download a copy of the performance report in PDF or HTML format, select the more actions icon <img alt="More actions icon" src="https://assets.postman.com/postman-docs/icon-more-actions-v9.jpg#icon" width="16px"> and select **Export PDF report** or **Export HTML report**. The report shows response time and throughput, slowest requests, requests with the most errors, metrics for each request, and a list of errors.
+* To view a list of all past performance runs for the collection, select the more actions icon <img alt="More actions icon" src="https://assets.postman.com/postman-docs/icon-more-actions-v9.jpg#icon" width="16px"> and select **View all runs**.
+* To open a request, select the name of a request.
+
+## Customizing the performance view
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/performance-test-metrics.md:51]:
```suggestion
## Customize the performance view
```

File: src/pages/docs/collections/performance-testing/testing-api-performance.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+The performance test runs on your computer and not in the cloud. When you start the test, the Collection Runner displays performance metrics in real time. You can [view performance test metrics](/docs/collections/performance-testing/performance-test-metrics/) such as the average response time, error rate, and throughput for all requests or individual requests. You can also [view details about errors](/docs/collections/performance-testing/performance-test-errors/) that occurred during the test to help identify the source of any performance issues.
+
+## Contents
+
+* [Configuring a performance test](#configuring-a-performance-test)
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/testing-api-performance.md:29]:
```suggestion
* [Configure a performance test](#configure-a-performance-test)
```

File: src/pages/docs/collections/performance-testing/testing-api-performance.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+* [Virtual users and system resources](#virtual-users-and-system-resources)
+* [Viewing past performance runs](#viewing-past-performance-runs)
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/testing-api-performance.md:31]:
```suggestion
* [View past performance runs](#view-past-performance-runs)
```

File: src/pages/docs/collections/performance-testing/testing-api-performance.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+## Configuring a performance test
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/testing-api-performance.md:33]:
```suggestion
## Configure a performance test
```

File: src/pages/docs/collections/performance-testing/testing-api-performance.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+Use the following guidelines to understand how many virtual users you can simulate:
+
+* A host with 8 CPU cores and 16 GB RAM can simulate up to 250 virtual users
+* A host with 16 CPU cores and 32 GB RAM can simulate up to 500 virtual users
+
+Attempting to simulate a higher number of virtual users may cause inaccurate metrics and reduced throughput (requests per second). Postman recommends running a small test with 10 to 20 virtual users and observing system resource usage. You can use this information to estimate that maximum number of virtual users that can be simulated for your collection and host computer.
+
+> If you have a problem with the performance testing feature, contact the [Postman support team](https://support.postman.com/hc/en-us). Make sure to attach your [Postman logs](https://support.postman.com/hc/en-us/articles/360025298633-How-to-get-logs-from-the-Postman-Desktop-app) in your support request.
+
+## Viewing past performance runs
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/testing-api-performance.md:89]:
```suggestion
## View past performance runs
```

File: src/pages/docs/collections/performance-testing/testing-api-performance.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+## Virtual users and system resources
+
+The number of virtual users a performance test can simulate depends on available system resources and the collection used for the test. Using pre-request or test scripts will reduce the number of virtual users that can be simulated. The maximum number of virtual users is determined by your [Postman plan](https://www.postman.com/pricing/).
+
+Use the following guidelines to understand how many virtual users you can simulate:
+
+* A host with 8 CPU cores and 16 GB RAM can simulate up to 250 virtual users
+* A host with 16 CPU cores and 32 GB RAM can simulate up to 500 virtual users
+
+Attempting to simulate a higher number of virtual users may cause inaccurate metrics and reduced throughput (requests per second). Postman recommends running a small test with 10 to 20 virtual users and observing system resource usage. You can use this information to estimate that maximum number of virtual users that can be simulated for your collection and host computer.
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/performance-testing/testing-api-performance.md:85]:
```suggestion
Attempting to simulate a higher number of virtual users may cause inaccurate metrics and reduced throughput (requests per second). Try running a small test with 10 to 20 virtual users and observing system resource usage. You can use this information to estimate that maximum number of virtual users that can be simulated for your collection and host computer.
```

File: src/pages/docs/collections/running-collections/working-with-data-files.md [hunk 12]
```diff
@@ -23,19 +23,25 @@ contextual_links:
...
 
 <img alt="CSV format" src="https://assets.postman.com/postman-docs/v10/ramen-csv-screenshot.jpg" width="300px"/>
 
 For CSV files, note the following:
 
-* If your CSV file contains numbers with preceding zeroes (for example, `000000345`) or phone numbers (for example, `+12125556709`), you'll need to specify the column type in **Preview**. Learn how in the next section.
+* If the numbers in your data file are longer than 15 digits, you'll need to format them as text in your spreadsheet program, so they're not truncated when exporting to CSV format.
 
-* If the numbers in your data file are longer than 15 digits, you'll need to format them as text in your spreadsheet program, so they're not truncated during import.
+* If your CSV file has numbers longer than 15 digits, numbers with preceding zeroes (for example, `000000345`), or phone numbers (for example, `+12125556709`), you'll need to preview the file and specify the column type. Learn how in the next section.
```
Comment (nerminamiller-postman) [inline src/pages/docs/collections/running-collections/working-with-data-files.md:42]:
```suggestion
* If your CSV file has numbers with preceding zeroes (for example, `000000345`), or phone numbers (for example, `+12125556709`), you'll need to preview the file and specify the column type. Learn how in the next section.
```

**Review by nerminamiller-postman — APPROVED** [review]
Very nice work, @mark-anthony-postman! I left just a few suggestions for you to consider.

File: src/pages/docs/collections/performance-testing/performance-test-data-files.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+## Contents
+
+* [Data file format](#data-file-format)
+* [Run a performance test with a data file](#run-a-performance-test-with-a-data-file)
+* [Debug errors when using data files](#debug-errors-when-using-data-files)
+
+## Data file format
+
+Format the CSV file so that the first row has the variable names you want to use inside the requests. After that, every row will be used as a data row. The line endings of the CSV file must be in Unix format. Each row must have the same number of columns.
```
Comment (postfrzero) [inline src/pages/docs/collections/performance-testing/performance-test-data-files.md:31]: I think it'll be helpful to mention whether or not the variable names respect the casing of the headings, and we should mention where these variables will be resolved. + @shreyas-panduranga Can you elaborate on both of these?

File: src/pages/docs/collections/performance-testing/performance-test-errors.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+* [Viewing top errors](#viewing-top-errors)
+* [Viewing error trends](#viewing-error-trends)
+* [Debugging errors](#debugging-errors)
+
+## Viewing top errors
+
+You can view top errors on the **Summary** tab while the performance test is running or after the test is complete.
+
+Hover over a point on the graph to view the top three errors at that point.
```
[the remark this answers is not in the document]
Comment (mark-anthony-postman) [reply src/pages/docs/collections/performance-testing/performance-test-errors.md:31]: We use "hover over" in the LC, which is acceptable per Microsoft: https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/mouse-mouse-interaction-terms

That said, this is certainly something we can consider, so you might want to add it to the style guide backlog

File: src/pages/docs/collections/running-collections/working-with-data-files.md (hunk 12 above)
Comment (mark-anthony-postman) [reply src/pages/docs/collections/running-collections/working-with-data-files.md:42]: So I tested this out, and you do need to specify `string` column type for long numbers when previewing in Postman. It's actually for numbers longer than 16, not 15. But what happens in Postman is that the long number is not actually truncated, but it gets rounded. So for example `123456789012345678` becomes `123456789012345680` when previewing. But selecting `string` fixes it. Give it a test yourself and see if you can confirm.

File: src/pages/docs/collections/performance-testing/performance-test-data-files.md (hunk 13 above)
Comment (mark-anthony-postman) [reply src/pages/docs/collections/performance-testing/performance-test-data-files.md:31]: I think when I was testing it did seem like the headings/variable names were case sensitive. Not sure what you mean by "where these variables will be resolved". @shreyas-panduranga can you clarify? Thanks!

**Review by petewitcher-postman — APPROVED** [review]
LGTM!

File: src/pages/docs/collections/performance-testing/performance-test-data-files.md (hunk 13 above)
Comment (shreyas-panduranga) [reply src/pages/docs/collections/performance-testing/performance-test-data-files.md:31]: @mark-anthony-postman Yes, the variables are case sensitive. And, the variables are resolved/available as local variables in the [Postman variables scope levels](https://learning.postman.com/docs/sending-requests/variables/#variable-scopes). So, they will be able to fetch a value coming from the data-file like this: `pm.variables.get('<variable-name-in-the-same-case-as-in-file>')`.

File: src/pages/docs/collections/performance-testing/performance-test-data-files.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+    url: "https://blog.postman.com/postman-api-performance-testing/"
+---
+
+When you configure a [performance test](/docs/collections/performance-testing/testing-api-performance/), you specify a number of _virtual users_. During the test, each virtual user (VU) runs the selected requests in the specified order in a repeating loop.
+
+By default, each virtual user runs the same requests using the same values. You can customize the behavior of virtual users by importing a CSV or JSON file. Each virtual user can send requests using different values from the imported data file. This enables you to better simulate real-world traffic when testing the performance of your API.
+
+## Contents
+
+* [Data file format](#data-file-format)
```
Comment (malvikach) [inline src/pages/docs/collections/performance-testing/performance-test-data-files.md:25]: @mark-anthony-postman I think "Running performance test with data files" should be the first topic on this page.

File: src/components/LeftNav/LeftNavItems.jsx [hunk 16]
```diff
@@ -542,7 +542,26 @@ export const leftNavItems = [
...
+          {
+            name: 'View performance test metrics',
+            url: '/docs/collections/performance-testing/performance-test-metrics/',
+          },
+          {
+            name: 'Debug performance test errors',
+            url: '/docs/collections/performance-testing/performance-test-errors/',
+          },
+          {
+            name: 'Import data for virtual users',
```
[the remark this answers is not in the document]
Comment (mark-anthony-postman) [reply src/components/LeftNav/LeftNavItems.jsx:561]: I changed the title in the left navigation menu to "Inject data into virtual users"

File: src/pages/docs/collections/performance-testing/performance-test-data-files.md [hunk 17]
```diff
@@ -0,0 +1,85 @@
+---
+title: "Use imported data to simulate virtual users"
```
[the remark this answers is not in the document]
Comment (mark-anthony-postman) [reply src/pages/docs/collections/performance-testing/performance-test-data-files.md:2]: I changed the page title to: "Use a data file to simulate virtual users"

**Review by malvikach — APPROVED** [review]

**Review by jkonrath-postman — APPROVED** [review]
LGTM

**Pull request merged**